label_studio/data_manager/api.py

```diff
@@ -163,8 +168,39 @@ def reset(self, request):
         queryset.all().delete()
         return Response(status=204)
 
+    @swagger_auto_schema(
+        method='put',
+        tags=['Data Manager'],
+        operation_summary='Update order of views',
+        operation_description='Update the order field of views based on the provided list of view IDs',
+        request_body=ViewOrderSerializer,
+    )
+    @action(detail=False, methods=['put'], url_path='order')
+    def update_order(self, request):
+        serializer = ViewOrderSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        project_id = serializer.validated_data['project']
+        view_ids = serializer.validated_data['ids']
+
+        project = generics.get_object_or_404(Project.objects.for_user(request.user), pk=project_id)
+
+        queryset = self.filter_queryset(self.get_queryset()).filter(project=project)
+        views = list(queryset.filter(id__in=view_ids))
```

Comment:
[select_for_update()](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#select-for-update) may be useful here to make sure that the orders being updated are exactly the same as the ones we juts fetched

Reply:
I'm not sure we need it here. I update only `order` field, in case of race condition of reordering requests, we will get "last db request win" situation anyway. 
If some other user want's to change filters(for example) of one particular tab, why do we need to deny it by locking the record?

Reply:
Yeah that's a very good point. It would be over-locking for non-sorting/non-conflict causing updates. We could normalize the ordering into a different table to get around this, but that's probably way too excessive for what would likely be a very unlikely edge case.